### PR TITLE
fixes #3976: drop ext/ctype dependency

### LINKED ISSUE #3976: drop ext/ctype dependency
requiring less dependencies is a good thing for a fundamental tool like composer is.

what do you think about rewriting stuff to "simple" sanity checks instead of relying on ext/ctype?
could provide a PR if you agree with it.

### PR DESCRIPTION
**Pull request by marc-mabe**
#3976

### CODE REVIEW

File: src/Composer/Package/Loader/ArrayLoader.php [hunk 1]
```diff
@@ -147,7 +147,7 @@ public function load(array $config, $class = 'Composer\Package\CompletePackage')
         }
 
         if (!empty($config['time'])) {
-            $time = ctype_digit($config['time']) ? '@'.$config['time'] : $config['time'];
+            $time = preg_match('/^[0-9]$/m', $config['time']) ? '@'.$config['time'] : $config['time'];
```
Comment (Seldaek) [inline src/Composer/Package/Loader/ArrayLoader.php]: Can you drop the `m` modifier here? I don't see how it makes sense

**Comment by Seldaek** [comment]
I guess I am fine with it given it's one line, but can you also PR the installer file in https://github.com/composer/getcomposer.org then? It also needs to be updated.

File: src/Composer/Package/Loader/ArrayLoader.php (hunk 1 above)
Comment (marc-mabe) [inline src/Composer/Package/Loader/ArrayLoader.php]: You are right, the `m` modifier is wrong. What I mean was the `D` modifier.
-> see http://3v4l.org/chkLF

**Comment by marc-mabe** [comment]
@Seldaek done composer/getcomposer.org#80

**Comment by Seldaek** [comment]
Thanks!

**Pull request merged**